src/ledger/LedgerTxn.cpp

```diff
@@ -168,11 +169,98 @@ LedgerEntryPtr::isDeleted() const
 {
     return mState == EntryPtrState::DELETED;
 }
+bool
+LedgerKeyMeter::canLoad(LedgerKey const& key, size_t entrySizeBytes) const
+{
+    return maxReadQuotaForKey(key) >= entrySizeBytes;
+}
+
+void
+LedgerKeyMeter::addTxn(SorobanResources const& resources)
+{
+    mTxReadBytes.push_back(resources.readBytes);
+    auto txId = mTxReadBytes.size() - 1;
+    auto addKeyToTxnMap = [&](auto const& key) {
+        auto iter = mLedgerKeyToTxs.find(key);
```

Comment:
Why not `mLedgerKeyToTxs[key].insert(txId)`? Also, why `set` and not `vector`?

Reply:
Hmmm I avoided `[]` as per Garand's recommendations elsewhere, but actually its fine here. (He also said that https://github.com/stellar/stellar-core/pull/4178#discussion_r1566258694) I just missed it. 

This doesn't need to be a `set`, I'll change both. :)